src/util/errors/RedirectHttpError.ts

```diff
@@ -1,19 +1,51 @@
-import type { HttpErrorOptions } from './HttpError';
-import { HttpError } from './HttpError';
+import type { HttpErrorClass, HttpErrorOptions } from './HttpError';
+import { generateHttpErrorClass, HttpError } from './HttpError';
 
 /**
- * Abstract class representing a 3xx redirect.
+ * An error corresponding to a 3xx status code.
+ * Includes the location it redirects to.
  */
-export abstract class RedirectHttpError extends HttpError {
-  public readonly location: string;
+export interface RedirectHttpError<TCode extends number = number> extends HttpError<TCode> {
+  readonly location: string;
+}
+
+/**
+ * Checks if the given error is a {@link RedirectHttpError}.
+ */
+export function isRedirectHttpError(error: any): error is RedirectHttpError {
```

Comment:
Why is this one inconsistent with all the rest?
I think we really need to work that difference away.

Reply:
Problem is that `RedirectHttpError` is an interface now because the generated redirect errors already extend the dynamic class that is created there, so they can't also extend `RedirectHttpError`. But I'm thinking now `RedirectHttpError` could still be a class, and the actual instances can implement that class instead of extending it, as that also works. Then we can have the `isInstance` function back there.